lib/osf-components/addon/components/cedar-metadata-renderer/component.ts

```diff
@@ -18,6 +19,15 @@ export default class CedarMetadataRenderer extends Component<Args> {
     viewerConfig = config.cedarConfig.viewerConfig;
     @tracked isShowEditor = false;
 
+    public get hasWritePermission(): boolean {
+        const node = this.args.cedarMetadataRecord.target as NodeModel;
+        if (node.get('modelName') === 'file') {
+            return (node as any).get('target').get('userHasWritePermission');
+        } else {
+            return node.get('userHasWritePermission');
+        }
```

Comment:
I think the names/types here are a bit confusing/misleading. It's a little bit more verbose, but I think these names and types are a bit more accurate/clear
```suggestion
        const target = this.args.cedarMetadataRecord.target as NodeModel | FileModel;
        if (target.get('modelName') === 'file') {
            return (target as FileModel).get('target').get('userHasWritePermission');
        } else {
            return (target as NodeModel).get('userHasWritePermission');
        }
```